keras_cv/models/darknet.py

```diff
@@ -0,0 +1,258 @@
+        include_rescaling: whether or not to Rescale the inputs.If set to True,
+            inputs will be passed through a `Rescaling(1/255.0)` layer.
+        include_top: whether to include the fully-connected layer at the top of
+            the network.  If provided, `num_classes` must be provided.
+        num_classes: optional number of classes to classify imagesinto, only to be
+            specified if `include_top` is True, and if no `weights` argument is
+            specified.
+        weights: one of `None` (random initialization), or a pretrained weight
+            file path.
+        input_shape: optional shape tuple, defaults to (None, None, 3).
+        pooling: optional pooling mode for feature extraction when `include_top`
+            is `False`.
+            - `None` means that the output of the model will be the 4D tensor output
+                of the last convolutional block.
+            - `avg` means that global average pooling will be applied to the
+                output of the last convolutional block, and thus the output of the
+                model will be a 2D tensor.
+            - `max` means that global max pooling will be applied.
+        classifier_activation: A `str` or callable. The activation function to use
+            on the "top" layer. Ignored unless `include_top=True`. Set
+            `classifier_activation=None` to return the logits of the "top" layer.
+            When loading pretrained weights, `classifier_activation` can only
+            be `None` or `"softmax"`.
+        name: (Optional) name to pass to the model.  Defaults to "DarkNet".
+    Returns:
+        A `keras.Model` instance.
+    """
+    if weights and not tf.io.gfile.exists(weights):
+        raise ValueError(
+            "The `weights` argument should be either `None` or the path to the "
+            f"weights file to be loaded. Weights file not found at location: {weights}"
+        )
+
+    if include_top and not num_classes:
+        raise ValueError(
+            "If `include_top` is True, you should specify `num_classes`. Received: "
+            f"num_classes={num_classes}"
+        )
+
+    inputs = layers.Input(shape=input_shape)
+
+    x = inputs
+    if include_rescaling:
+        x = layers.Rescaling(1 / 255.0)(x)
+
+    # stem
+    x = DarknetConvBlock(
+        filters=32, kernel_size=3, strides=1, activation="leaky_relu", name="stem_conv"
+    )(x)
+    x = ResidualBlocks(filters=64, num_blocks=1, name="stem_residual_block")(x)
+
+    # dark2
+    x = ResidualBlocks(filters=128, num_blocks=blocks[0], name="dark2_residual_block")(
+        x
+    )
+
+    # dark3
+    x = ResidualBlocks(filters=256, num_blocks=blocks[1], name="dark3_residual_block")(
+        x
+    )
+
+    # dark4
+    x = ResidualBlocks(filters=512, num_blocks=blocks[2], name="dark4_residual_block")(
+        x
+    )
+
+    # dark5
+    x = ResidualBlocks(filters=1024, num_blocks=blocks[3], name="dark5_residual_block")(
```

Comment:
should this instead be `for filters, blocks in zip(filters, blocks)` with some rewriting to make this possible?

Reply:
Should be doable, I'll try updating this.